uxarray/remap/inverse_distance_weighted.py

```diff
@@ -0,0 +1,297 @@
+    else:
+        raise ValueError(
+            f"Invalid source_data shape. The final dimension should match the number of corner "
+            f"nodes ({source_grid.n_node}), edge nodes ({source_grid.n_edge}), or face centers ({source_grid.n_face}) "
+            f"in the source grid, but received: {source_data.shape}"
+        )
+
+    if coord_type == "spherical":
+        if remap_to == "nodes":
+            lon, lat = (
+                destination_grid.node_lon.values,
+                destination_grid.node_lat.values,
+            )
+        elif remap_to == "face centers":
+            lon, lat = (
+                destination_grid.face_lon.values,
+                destination_grid.face_lat.values,
+            )
+        elif remap_to == "edge centers":
+            lon, lat = (
+                destination_grid.edge_lon.values,
+                destination_grid.edge_lat.values,
+            )
+        else:
+            raise ValueError(
+                f"Invalid remap_to. Expected 'nodes', 'edge centers', or 'face centers', "
+                f"but received: {remap_to}"
+            )
+
+        _source_tree = source_grid.get_ball_tree(coordinates=source_data_mapping)
+
+        dest_coords = np.vstack([lon, lat]).T
+
+        distances, nearest_neighbor_indices = _source_tree.query(
+            dest_coords, k=k_neighbors
+        )
+
+    elif coord_type == "cartesian":
+        if remap_to == "nodes":
+            x, y, z = (
+                destination_grid.node_x.values,
+                destination_grid.node_y.values,
+                destination_grid.node_z.values,
+            )
+        elif remap_to == "face centers":
+            x, y, z = (
+                destination_grid.face_x.values,
+                destination_grid.face_y.values,
+                destination_grid.face_z.values,
+            )
+        elif remap_to == "edge centers":
+            x, y, z = (
+                destination_grid.edge_x.values,
+                destination_grid.edge_y.values,
+                destination_grid.edge_z.values,
+            )
+        else:
+            raise ValueError(
+                f"Invalid remap_to. Expected 'nodes', 'edge centers', or 'face centers', "
+                f"but received: {remap_to}"
+            )
+
+        _source_tree = source_grid.get_kd_tree(coordinates=source_data_mapping)
+
+        dest_coords = np.vstack([x, y, z]).T
+
+        distances, nearest_neighbor_indices = _source_tree.query(
+            dest_coords, k=k_neighbors
+        )
+
+    else:
+        raise ValueError(
+            f"Invalid coord_type. Expected either 'spherical' or 'cartesian', but received {coord_type}"
+        )
+
+    if nearest_neighbor_indices.ndim > 1:
+        nearest_neighbor_indices = nearest_neighbor_indices.squeeze()
+
+    weights = []
+    for ind in range(len(distances)):
+        weight = 1 / (distances[ind] ** power + 1e-6)
+        weight /= np.sum(weight)
+        weights.append(weight)
```

Comment:
Does this need to be in a for loop? I'd suggest trying to vectorize it. Should be able to do some tricks using `axis=1` and some numpy functionality. 

Let me know if you need any help!

Reply:
@philipc2 Oh yeah, you are right, I can use my original code for the weights if I use an axis on the sum. Thanks for pointing that out!